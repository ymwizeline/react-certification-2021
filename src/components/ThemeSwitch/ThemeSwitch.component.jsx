import React from 'react';
import Switch from '@material-ui/core/Switch';
import { Container } from './styled';
import SunIcon from '../Icons/SunIcon';
import MoonIcon from '../Icons/MoonIcon';

const ThemeSwitch = () => {
  return (
    <Container>
      <MoonIcon />
      <Switch
        defaultChecked
        color="default"
        inputProps={{ 'aria-label': 'checkbox with default color' }}
      />
      <SunIcon />
    </Container>
  );
};

export default ThemeSwitch;
