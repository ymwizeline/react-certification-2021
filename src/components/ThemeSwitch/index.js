export { default } from './ThemeSwitch.component';
