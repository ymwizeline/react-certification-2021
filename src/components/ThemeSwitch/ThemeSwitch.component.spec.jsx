import React from 'react';
import renderer from 'react-test-renderer';
import ThemeSwitch from './ThemeSwitch.component';
import 'jest-styled-components';

describe('ThemeSwitch component', () => {
  it(`renders properly`, () => {
    const element = renderer.create(<ThemeSwitch />).toJSON();

    expect(element).toMatchSnapshot();
  });
});
