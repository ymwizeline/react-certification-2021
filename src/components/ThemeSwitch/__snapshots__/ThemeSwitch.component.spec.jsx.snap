// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch component renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <svg
    className=""
    enableBackground="new 0 0 24 24"
    height="20px"
    size={20}
    viewBox="0 0 24 24"
    width="20px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M0,0h24v24H0V0z"
        fill="none"
      />
    </g>
    <g>
      <path
        d="M9.5,2c-1.82,0-3.53,0.5-5,1.35c2.99,1.73,5,4.95,5,8.65s-2.01,6.92-5,8.65C5.97,21.5,7.68,22,9.5,22c5.52,0,10-4.48,10-10 S15.02,2,9.5,2z"
      />
    </g>
  </svg>
  <span
    className="MuiSwitch-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase PrivateSwitchBase-checked-2 Mui-checked"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          aria-label="checkbox with default color"
          className="PrivateSwitchBase-input-4 MuiSwitch-input"
          defaultChecked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="MuiSwitch-thumb"
        />
      </span>
    </span>
    <span
      className="MuiSwitch-track"
    />
  </span>
  <svg
    className=""
    enableBackground="new 0 0 24 24"
    height="25px"
    size={25}
    viewBox="0 0 24 24"
    width="25px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91l-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
    />
  </svg>
</div>
`;
